File: arrays/group_anagrams/group_anagrams.py
""" Solution Module """
class Solution:
    """ Solution Class """
    def run_solution(self, test_input):
        """ Solution Method """
        # // Input: strs = ["act","pots","tops","cat","stop","hat"]
        # // Output: [["hat"],["act", "cat"],["stop", "pots", "tops"]]
        anagram_dict: dict[str, list[str]] = dict()

        for i in test_input:
            sorted_list = "".join(sorted(i))
            if sorted_list in anagram_dict:
                anagram_dict[sorted_list].append(i)
            else :
                anagram_dict[sorted_list] = [i]
        return anagram_dict

File: arrays/group_anagrams/group_anagrams.ts
import { hash } from "node:crypto";
import * as test from "node:test";

export class Solution {
    /**
    * @param {string} testInput
    * @return {any} Returns either boolean to validate the test cases or a custom result. 
    */
    run_solution(testInput: any) {
        // Input: strs = ["act","pots","tops","cat","stop","hat"]
        // Output: [["hat"],["act", "cat"],["stop", "pots", "tops"]]

        const hashMap: Record<string, [string]> = {};
        
        testInput.forEach(element => {
            const sorted = element.split("").sort().join("");;
            if(hashMap[sorted]) {
                hashMap[sorted].push(element);
            } else {
                hashMap[sorted] = [element]
            }
        });
        return Object.values(hashMap)
    }
}